### DISCUSSION TITLE
How to implement a NodeOperation Handler for multiple elementIDs?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#644] [Q&A] [answered]
I wonder if it is possible to implement a CreateNodeOperation Handler for multiple elementIDs? 

I can see that the AbstractCreateOperationHandler provide such a constructor

```java 
public abstract class AbstractCreateOperationHandler<T extends CreateOperation> extends AbstractOperationHandler<T>
   implements CreateOperationHandler {
   protected List<String> handledElementTypeIds;

   public AbstractCreateOperationHandler(final String... elementTypeIds) {
      this(Lists.newArrayList(elementTypeIds));
   }
....
```

But the `CreateNodeOperationHandler` which is extending the `AbstractCreateOperationHandler` does not. 

As a result I implement now a CreateXXXOperationHandler (extending the `CreateNodeOperationHandler`) for each elementID in my diagram, even if the implementation is identically.

**Comment by planger** [comment, marked as answer]
You are right, I guess, we could also make [the constructor](https://github.com/eclipse-glsp/glsp-server/blob/39ee6c9aa37f71d989ca0efcfc49ae2f25690b3e/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/operations/gmodel/CreateNodeOperationHandler.java#L46) of  `CreateNodeOperationHandler` take a var arg and pass it to super, to allow for `CreateNodeOperationHandler` that can handle multiple element types.

I suggest to turn this into a GH issue. Feel free to open a PR! Seems like a good idea on the first sight.

**Reply by planger** [reply]
https://github.com/eclipse-glsp/glsp/issues/646
